### ISSUE TITLE
Heating and cooling season are not realistic

### ISSUE THREAD

**Issue by martin-mosteiro** [architecture-building-systems/CityEnergyAnalyst#2366]
The use of a single heating season and a single cooling season for the Swiss case does not appear to be realistic. For starters, it leads to buildings with high internal loads (e.g., `INDUSTRY`, `RETAIL`) to overheat in summer, as shown by @prakharmehta95 in #2221:
![image](https://user-images.githubusercontent.com/18548065/66301647-ce774580-e8f7-11e9-8742-598158599fb6.png)

The old version of SIA 2024 (2006) included monthly heating and cooling balances such as the following:
- Residential:
![Uploading Screen Shot 2019-10-07 at 11.45.42.png…]()
- Open office:
<img width="492" alt="Screen Shot 2019-10-07 at 11 46 36" src="https://user-images.githubusercontent.com/18548065/66301852-34fc6380-e8f8-11e9-8a25-629e40ae7ae9.png">
- Lecture hall:
<img width="494" alt="Screen Shot 2019-10-07 at 11 47 47" src="https://user-images.githubusercontent.com/18548065/66301901-4d6c7e00-e8f8-11e9-9a86-7945030b7820.png">
- Supermarket:
<img width="495" alt="Screen Shot 2019-10-07 at 11 48 42" src="https://user-images.githubusercontent.com/18548065/66302011-7987ff00-e8f8-11e9-93af-19159c6065db.png">
- Retail:
<img width="495" alt="Screen Shot 2019-10-07 at 11 48 54" src="https://user-images.githubusercontent.com/18548065/66302027-7ee54980-e8f8-11e9-8e22-f6d0d4d89bc4.png">

While all of these buildings appear to follow the rule of thumb (heating from mid-September to mid-May only), heating and cooling during the same month does appear to happen. Removing the cooling season limitation in CEA appears reasonable, then.

As discussed with @gabriel-happle, however, a building being warmed up in the morning when people arrive and being cooled a few hours later due to internal gains intuitively does not appear logical. Especially in seasons when outdoor temperatures are still cool (say, March). Any ideas are very welcome @gabriel-happle, @JIMENOFONSECA.

**Comment by gabriel-happle** [comment]
@martin-mosteiro I guess the rule to choose between heating and cooling operation should be based on something like the average daily outdoor temperature.

For residential it seems to be mostly a legal issue in Switzerland:
https://www.20min.ch/community/dossier/geldratgeber/story/Wann-muss-der-Vermieter-heizen--29988400

I don't know if the same applies to other building functions.

question: the retail cooling in the SIA graph space cooling or cooling of coolrooms, refrigeration, etc.?

**Comment by JIMENOFONSECA** [comment]
Sure, still this should be an option of the demand tool.

Just adding two variables will suffice. It would be up to anyone using the tool to set up this.

If you go for the dates, include:

config.demand.heating-season = 01.11, 30.03
config.demand.cooing-season = 01.04, 30.10

We can create with Reynold the interface for this. It can look like a calendar in CEA. only day and month are allowed for beginning and end of each season.

If you want the option of temperature_based heating season (what martin describes) then add a variable to it and add the code.

config.demand.temperature-based-heating-season =True

Maybe all two will be the best scenario.



On 8 Oct 2019, 2:08 PM +0800, Gabriel Happle <notifications@github.com>, wrote:

@martin-mosteiro<https://github.com/martin-mosteiro> I guess the rule to choose between heating and cooling operation should be based on something like the average daily outdoor temperature.

For residential it seems to be mostly a legal issue in Switzerland:
https://www.20min.ch/community/dossier/geldratgeber/story/Wann-muss-der-Vermieter-heizen--29988400

I don't know if the same applies to other building functions.

question: the retail cooling in the SIA graph space cooling or cooling of coolrooms, refrigeration, etc.?

**Comment by martin-mosteiro** [comment]
@gabriel-happle 

> For residential it seems to be mostly a legal issue in Switzerland

It is, but from what I gather the legal requirement is that building owners _have_ to provide heating when conditions reach X. That does _not_ mean, however, that they are required to shut down the heating outside of the heating season, does it?

> question: the retail cooling in the SIA graph space cooling or cooling of coolrooms, refrigeration, etc.?

Yeah, I wondered the same thing. I believe it is only space cooling. SIA 2024 (2006) says the _Jährlicher spezifischer Kühlenergiebedarf / Demande spécifique annuelle en énergie de refroidissement / Qc kWh/m2_ is defined according to SIA 380/4, which says the following:

> Kälteenergiebedarf – Demande en énergie pour le refroidissement – Qc – MJ/m2
> Kälteenergie, die über das Jahr notwendig ist, um die internen und externen Lasten bei konstanter Raumtemperatur abzuführen. Dies ist gleich der Jahressumme der internen und externen Lasten abzüglich der Wärmeverluste (Transmissionsverluste und Lüftungsverluste durch Infiltration), aufsummiert soweit die Differenz positiv ist.
> Zusätzlich umfasst der Kälteenergiebedarf die Kälte zur Kühlung des Aussenluftvolumenstroms auf die Raumluft-Solltemperatur inkl. der damit einhergehenden Entfeuchtung des Aussenluftvolumenstroms (Aussenluftkühlung; Jahressumme der Enthalpie, die dem Aussenluftstrom bei Abkühlung auf die Raumluft-Solltemperatur entzogen werden muss).

Nevertheless, SIA 2024 (2006) also states:
> Ein positiver Wert für den Kühlenergiebedarf bedeutet nicht, dass gekühlt werden muss. Der Bedarf für eine Kühlung richtet sich nach Norm SIA 382/1 Ziffer 4.4.

So I checked SIA 2024 (2015), which specifies whether a cooling system is required for each type of use as well as the necessary power and full load hours. For RETAIL (i.e., _Fachgeschäft_) a cooling system is required in all cases (_Kühlung mit Fensterlüftung Tag und Nacht: notwendig; Kühlung mit Fensterlüftung bei Belegung: notwending; Kühlung ohne Fensterlüftung: notwending_), with a cooling power of 41 W/m2 and 2160 full load hours (!).

So based on SIA 2024 (2015) RETAIL buildings definitely need cooling beyond the "heating season", even if natural ventilation is provided.

**Comment by martin-mosteiro** [comment]
@JIMENOFONSECA 
> Just adding two variables will suffice. It would be up to anyone using the tool to set up this.
> If you go for the dates, include:
> config.demand.heating-season = 01.11, 30.03
> config.demand.cooing-season = 01.04, 30.10
> If you want the option of temperature_based heating season (what martin describes) then add a variable to it and add the code.
> config.demand.temperature-based-heating-season =True

Actually, it's not the heating season I'm questioning, but the cooling season. i.e., the profiles found in SIA appear to match our assumed heating season, but _not_ our cooling season. So the answer would be to remove the cooling season definition entirely (or make the cooling season last all year) and adjust the code so that heating and cooling seasons overlap.

The tricky part, in my opinion, is getting a realistic definition of systems operation, so that buildings aren't cooled in unreasonable ways (like the office example given at the beginning).

**Comment by JIMENOFONSECA** [comment]
@martin-mosteiro I see, so in your case the building is overheating...I wonder if just changing the minimum ventilation per person (what might occur in reality) will help to not overheat the building. 

So how to proceed?

Is not this as easy as changing the start and end of the cooling season in the system's control file to make it last for the entire year? Should not that be all?

**Comment by martin-mosteiro** [comment]
Indeed, adjusting the heating/cooling season control is easy. The question is what would be a realistic building system operation for buildings with high internal gains such that cooling is required even at times where outdoor temperature is relatively low (such as RETAIL). Increasing ventilation sounds like a reasonably straightforward solution – I just don't know if that's indeed what the operators of such buildings actually do.

Given that if anything we seem to overpredict cooling in CEA as is, I think it's worth studying how such buildings are operated a bit further.

**Comment by JIMENOFONSECA** [comment]
Good. Any information online about this?
On 16 Oct 2019, 21:14 +0800, Martín Mosteiro-Romero <notifications@github.com>, wrote:

Indeed, adjusting the heating/cooling season control is easy. The question is what would be a realistic building system operation for buildings with high internal gains such that cooling is required even at times where outdoor temperature is relatively low (such as RETAIL). Increasing ventilation sounds like a reasonably straightforward solution – I just don't know if that's indeed what the operators of such buildings actually do.

Given that if anything we seem to overpredict cooling in CEA as is, I think it's worth studying how such buildings are operated a bit further.

**Comment by martin-mosteiro** [comment]
I think the solution is making night time flushing possible even beyond the cooling season. As a test, I ran the same retail building (ventilation system 'T01', i.e., night time flushing available) for a case where the cooling season was the normal cooling season in CEA, and the same case where the cooling season is from March to September. The results are shown below:
- **Case 1: current cooling season**
<img width="666" alt="Screen Shot 2019-10-17 at 11 12 00" src="https://user-images.githubusercontent.com/18548065/66998732-12770100-f0d5-11e9-8fcc-2bdcdfd7eb79.png">
No heating or cooling in spring and fall, buildings overheat to >30°C
- **Case 2: longer cooling season**
<img width="665" alt="Screen Shot 2019-10-17 at 11 52 47" src="https://user-images.githubusercontent.com/18548065/66998828-363a4700-f0d5-11e9-83ea-01d1e732fdb0.png">
Some cooling in spring and autumn, but the temperature never reaches even 25°C.

So I think making night time flushing available year-round, even without changing the cooling seasons, might solve our problem.

**Comment by martin-mosteiro** [comment]
Not sure what caused the summertime change in indoor temperature nor the drastic change in cooling demand though...

**Comment by martin-mosteiro** [comment]
Looking at @prakharmehta95's data, it seemed pretty clear that heat gains were accumulating with no way to release them. So I tested what would happen if we made economizers available year-round. This seems like a sensible system to have in buildings such as the ones @prakharmehta95 was having trouble with (`RETAIL`) which have massive internal gains and hence overheat even at times when outdoor temperatures are low.

After implementing economizers and allowing them to run year round, here is the indoor temperature in one of these problem buildings for both cases:
![Screen Shot 2019-10-22 at 21 18 49](https://user-images.githubusercontent.com/18548065/67322610-93892a80-f511-11e9-8b88-9ccd27c8e2b6.png)
The addition of the economizer reduces the peak temperature in the building by 5°C - which is already pretty good!

Adding an economizer reduces the cooling demand by 10%, which is probably okay since we seem to be overestimating cooling in the Swiss context in any case. More importantly, the unnatural sharp peak at the beginning of the cooling season is removed:
![Screen Shot 2019-10-22 at 21 33 20](https://user-images.githubusercontent.com/18548065/67324275-dc41e300-f513-11e9-9aab-0966b08c0723.png)

So I believe it makes sense to have economizers as the standard for retail buildings in Switzerland. Right now all ventilation systems in Switzerland are assumed to have no economizer – not sure if this is a realistic assumption or not.

Nevertheless, this adjustment alone doesn't solve the problem with retail buildings found in @prakharmehta95's case study, as even though the peak indoor temperature was reduced by 5°C, the indoor temperature does still reach 31.6°C in early May (specifically, on May 10th).

So the question remains: do we make it possible for cooling and heating seasons to overlap (and hence potentially have space heating and cooling on the same day? This is certainly not the case at the moment:
![Screen Shot 2019-10-22 at 21 53 37](https://user-images.githubusercontent.com/18548065/67326167-72770880-f516-11e9-8a8a-094a75a15ff6.png)

And according to the old SIA standard heating and cooling on the same month might indeed happen. Maybe adding the economizer as a first strategy and then allowing cooling if the temperature still exceeds the maximum threshold could be a sensible but still realistic solution.

Any thoughts @gabriel-happle?

**Comment by gabriel-happle** [comment]
I think they can overlap. I don't have any issue with that. We just need to come up with some smart way of controlling it. I.e., model some realistic "control system". Because what I am a bit scared of, is that cooling and heating might be activated alternating on the same day. E.g., heating at night and cooling in the afternoon on the same day...
I believe such an algorithm could be based on the difference between indoor and outdoor temperature. However, I don't have any specific source for that.

**Comment by JIMENOFONSECA** [comment]
I am not for overlaping since it adds some level complexity that a single zone building model should not handle (heating and cooling during the same hour). The best way is to regulate the ventilation rate or the control (night flush and economizer cycle). That may be more realistic. We can even think about a ventilation schedule. I wonder how Epllus does it, and if we can imitate some of that behavior.
On 24 Oct 2019, 20:46 +0800, Gabriel Happle <notifications@github.com>, wrote:

I think they can overlap. I don't have any issue with that. We just need to come up with some smart way of controlling it. I.e., model some realistic "control system". Because what I am a bit scared of, is that cooling and heating might be activated alternating on the same day. E.g., heating at night and cooling in the afternoon on the same day...
I believe such an algorithm could be based on the difference between indoor and outdoor temperature. However, I don't have any specific source for that.

**Comment by martin-mosteiro** [comment]
@JIMENOFONSECA Thanks for your feedback. Actually, I don't think cooling and heating in the same hour would ever happen, because the workflow would be something like:
1. If the temperature is too hot, use economizer/night flushing
2. Since temperature is too hot, no heating is activated
3. If temperature is still too hot, turn on cooling systems.

However, I think it _could_ be possible to have heating and cooling on the same day, which wouldn't be ideal (although that doesn't seem to be the case in the cases I tested so far). 

Furthermore, based on both SIA standards and my own measured data from ETH, it _does_ seem possible to have heating and cooling on the same month. So a top-down decision to forbid heating or cooling on certain months based solely on a predetermined season does not seem to match either data.

**Comment by martin-mosteiro** [comment]
@JIMENOFONSECA I'm not sure I understand the question, but in principle it's possible for the heating season to start at the middle of the month, yes.

**Comment by martin-mosteiro** [comment]
Instead of changing the cooling season duration, I have now implemented the following:
*1-Make heat recovery bypass available: *
So far we assumed this was not possible. Based on discussions with the Zurich team, this appears to be not only possible but common. Otherwise ventilation would have little effect on the indoor temperatures since the air being circulated in the building would be preheated by heat recovery.
*2-Make night flushing available year-round (so far it was only available during the cooling season)*
This seems like a reasonable assumption given that certain building occupancy types are overheating during the day according to the archetypal loads (in particular RETAIL). The effect on the reference case open is quite minor, though (see graphs below).
*3-Make economizers available year-round (so far they were only available during the cooling season)*
Again, this seems like a reasonable assumption: if the building is overheating during the day, it would make sense to increase ventilation to cool it down.
*4-Make economizers standard in new constructions*
The archetypes so far assumed most building occupancy types had a ventilation system type 'T1', meaning mechanical ventilation with heat recovery and night flushing, but no economizer. This seems to be insufficient for RETAIL buildings given the loads defined by SIA standards. Making ventilation available when the indoor temperature gets too hot intuitively does not seem like such a complicated approach. So I assumed all buildings built or renovated after 2006 would have ventilation systems type 'T2' (i.e., same as 'T1' but with economizers).
*5-Make future residential buildings have ventilation systems*
Since Minergie buildings require mechanical ventilation, it seems like a reasonable assumption that residential buildings built after 2021 would have mechanical ventilation. So I added this to the archetypes database.

I tested the aforementioned changes on `reference-case-open` and found the following.

*B01 - 75% FOODSTORE, 25% COOLROOM*
The internal temperature in intermediate seasons reaches ~28ºC when running the current CEA simulation (no night flushing or economizers during the heating season) as well as when night flushing is indeed available year-round. When an economizer is added, the peak temperature in the intermediate seasons decreases by about 1ºC.

The same is true about the heating and cooling demands throughout the year. The availability of an economizer reduces the peak cooling demand by about 20%.

*B06 - 50% RETAIL, 50% RESTAURANT*
The peak temperature in autumn reaches 29.8ºC. The addition of night flushing in spring reduces the peaks somewhat, but not in autumn. The addition of an economizer keeps indoor temperatures below 27ºC year-round.

The heating demands are unchanged by the change in ventilation controls, whereas cooling demand is decreased significantly by the economizer. The peak demand also decreases by about 30%.

Looking at the yearly demands of all buildings, the heating demand is practically unchanged by night flushing, whereas the heating demand changes by 0-2% with the economizer (average: +0.1%). The cooling demand on the other hand goes up very slightly due to night flushing (+0.2% on average) whereas the cooling demand goes down significantly when the economizer is available (-27%-+3%, average -12.5%). The electricity demand is unchanged by night flushing and goes down by 1.2% due to the economizer.

I'm creating a PR with these changes and assigning @JIMENOFONSECA as a reviewer.

**Comment by JIMENOFONSECA** [comment]
Thanks for the comprehensive description martin. I think this may solve many of the issues we had. If I get it right. These properties become part of the archetypes right?


On 8 Nov 2019, 12:54 AM +0800, Martín Mosteiro-Romero <notifications@github.com>, wrote:

Instead of changing the cooling season duration, I have now implemented the following:
*1-Make heat recovery bypass available: *
So far we assumed this was not possible. Based on discussions with the Zurich team, this appears to be not only possible but common. Otherwise ventilation would have little effect on the indoor temperatures since the air being circulated in the building would be preheated by heat recovery.
2-Make night flushing available year-round (so far it was only available during the cooling season)
This seems like a reasonable assumption given that certain building occupancy types are overheating during the day according to the archetypal loads (in particular RETAIL). The effect on the reference case open is quite minor, though (see graphs below).
3-Make economizers available year-round (so far they were only available during the cooling season)
Again, this seems like a reasonable assumption: if the building is overheating during the day, it would make sense to increase ventilation to cool it down.
4-Make economizers standard in new constructions
The archetypes so far assumed most building occupancy types had a ventilation system type 'T1', meaning mechanical ventilation with heat recovery and night flushing, but no economizer. This seems to be insufficient for RETAIL buildings given the loads defined by SIA standards. Making ventilation available when the indoor temperature gets too hot intuitively does not seem like such a complicated approach. So I assumed all buildings built or renovated after 2006 would have ventilation systems type 'T2' (i.e., same as 'T1' but with economizers).
5-Make future residential buildings have ventilation systems
Since Minergie buildings require mechanical ventilation, it seems like a reasonable assumption that residential buildings built after 2021 would have mechanical ventilation. So I added this to the archetypes database.

I tested the aforementioned changes on reference-case-open and found the following.

B01 - 75% FOODSTORE, 25% COOLROOM
The internal temperature in intermediate seasons reaches ~28ºC when running the current CEA simulation (no night flushing or economizers during the heating season) as well as when night flushing is indeed available year-round. When an economizer is added, the peak temperature in the intermediate seasons decreases by about 1ºC.

The same is true about the heating and cooling demands throughout the year. The availability of an economizer reduces the peak cooling demand by about 20%.

B06 - 50% RETAIL, 50% RESTAURANT
The peak temperature in autumn reaches 29.8ºC. The addition of night flushing in spring reduces the peaks somewhat, but not in autumn. The addition of an economizer keeps indoor temperatures below 27ºC year-round.

The heating demands are unchanged by the change in ventilation controls, whereas cooling demand is decreased significantly by the economizer. The peak demand also decreases by about 30%.

Looking at the yearly demands of all buildings, the heating demand is practically unchanged by night flushing, whereas the heating demand changes by 0-2% with the economizer (average: +0.1%). The cooling demand on the other hand goes up very slightly due to night flushing (+0.2% on average) whereas the cooling demand goes down significantly when the economizer is available (-27%-+3%, average -12.5%). The electricity demand is unchanged by night flushing and goes down by 1.2% due to the economizer.

I'm creating a PR with these changes and assigning @JIMENOFONSECA<https://github.com/JIMENOFONSECA> as a reviewer.
